### PR TITLE
feat: Add torch compile

### PR DESCRIPTION
**Pull request by yuanzhedong**

### CODE REVIEW

**Comment by yuanzhedong** [comment]
/ok to test [3c3a225](https://github.com/NVIDIA-NeMo/Automodel/pull/318/commits/3c3a225bad32b6433cbc2a20313e5a1565ad3be4)

**Comment by yuanzhedong** [comment]
/ok to test [82ada52](https://github.com/NVIDIA-NeMo/Automodel/pull/318/commits/82ada5247e882b92593f393f4c22b3c594707dd0)

**Comment by yuanzhedong** [comment]
/ok to test [b098c61](https://github.com/NVIDIA-NeMo/Automodel/pull/318/commits/b098c619582d9c08c441ae27b35a006bd98e6da6)

**Comment by yuanzhedong** [comment]
/ok to test [39858e2](https://github.com/NVIDIA-NeMo/Automodel/pull/318/commits/39858e2dab2c57a4fdcfe79c31993a92252f7517)

**Comment by yuanzhedong** [comment]
/ok to test [8c76190](https://github.com/NVIDIA-NeMo/Automodel/pull/318/commits/8c761903a41d92c849262dd22ff1e6ec4d63ac3b)

**Comment by yuanzhedong** [comment]
/ok to test [8c76190](https://github.com/NVIDIA-NeMo/Automodel/pull/318/commits/8c761903a41d92c849262dd22ff1e6ec4d63ac3b)

**Comment by yuanzhedong** [comment]
/ok to test [4fe6a26](https://github.com/NVIDIA-NeMo/Automodel/pull/318/commits/4fe6a264a755987233889dc1bf89b80559e71ee5)

**Comment by yuanzhedong** [comment]
/ok to test [4960f5d](https://github.com/NVIDIA-NeMo/Automodel/pull/318/commits/4960f5d7bcf11d35056aa23322abdb32a5b075cf)

**Review by HuiyingLi — APPROVED** [review]
LGTM, thank you so much!

**Comment by yuanzhedong** [comment]
/ok to test [b4bd3aa](https://github.com/NVIDIA-NeMo/Automodel/pull/318/commits/b4bd3aabeffd6c94a666166b30c09763c8af1459)

File: nemo_automodel/components/utils/compile_utils.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        "dynamic": config.dynamic,
+    }
+    if config.backend is not None:
+        compile_kwargs["backend"] = config.backend
+    compile_kwargs.update(options_dict)
+
+    logger.info(f"Compiling model with backend={config.backend}, mode={config.mode}, dynamic={config.dynamic}")
+
+    try:
+        compiled_model = torch.compile(model, **compile_kwargs)
```
Comment (akoumpa) [inline nemo_automodel/components/utils/compile_utils.py:205]:
```suggestion
        model.compile(**compile_kwargs)
```
WDYT about using the `compile` method that is available in the `nn.Module` class? The advantage I see is not modifying the FQN in the state_dict.
